deploy/charts/cert-manager/values.yaml

```diff
@@ -1,251 +1,73 @@
-# Default values for cert-manager.
```

Comment:
This file too would benefit from a header comment explaining that it is auto-generated by xxx 

*AND* retain the comment that explains that it contains the default values for the Helm chart.

I checked this file by using `yq` to sort all the keys in this and the original and then doing a diff. 
I modified the `values/` source files so that there are no unnecessary empty slices (`[]`) or maps `{}` in here.

```sh
diff --color --ignore-all-space -u \
  <(curl -sSL https://raw.githubusercontent.com/jetstack/cert-manager/master/deploy/charts/cert-manager/values.yaml | bazel run //hack/bin:yq -- eval --tojson 'sortKeys(..)' - ) \
  <(bazel run //hack/bin:yq -- eval --tojson 'sortKeys(..)' deploy/charts/cert-manager/values.yaml)
```


```diff
--- /dev/fd/63	2021-09-10 16:06:30.385954195 +0100
+++ /dev/fd/62	2021-09-10 16:06:30.386954211 +0100
@@ -1,35 +1,20 @@
 {
-  "affinity": {},
   "cainjector": {
-    "affinity": {},
-    "containerSecurityContext": {},
     "enabled": true,
-    "extraArgs": [],
     "image": {
       "pullPolicy": "IfNotPresent",
       "repository": "quay.io/jetstack/cert-manager-cainjector"
     },
-    "nodeSelector": {},
-    "podLabels": {},
     "replicaCount": 1,
-    "resources": {},
     "securityContext": {
       "runAsNonRoot": true
     },
     "serviceAccount": {
       "automountServiceAccountToken": true,
       "create": true
+    }
     },
-    "strategy": {},
-    "tolerations": []
-  },
-  "clusterResourceNamespace": "",
-  "containerSecurityContext": {},
-  "extraArgs": [],
-  "extraEnv": [],
-  "featureGates": "",
   "global": {
-    "imagePullSecrets": [],
     "leaderElection": {
       "namespace": "kube-system"
     },
@@ -38,7 +23,6 @@
       "enabled": false,
       "useAppArmor": true
     },
-    "priorityClassName": "",
     "rbac": {
       "create": true
     }
@@ -49,14 +33,11 @@
   },
   "ingressShim": {},
   "installCRDs": false,
-  "nodeSelector": {},
-  "podLabels": {},
   "prometheus": {
     "enabled": true,
     "servicemonitor": {
       "enabled": false,
       "interval": "60s",
-      "labels": {},
       "path": "/metrics",
       "prometheusInstance": "default",
       "scrapeTimeout": "30s",
@@ -64,7 +45,6 @@
     }
   },
   "replicaCount": 1,
-  "resources": {},
   "securityContext": {
     "runAsNonRoot": true
   },
@@ -73,10 +53,8 @@
     "create": true
   },
   "startupapicheck": {
-    "affinity": {},
     "backoffLimit": 4,
     "enabled": true,
-    "extraArgs": [],
     "image": {
       "pullPolicy": "IfNotPresent",
       "repository": "quay.io/jetstack/cert-manager-ctl"
@@ -86,8 +64,6 @@
       "helm.sh/hook-delete-policy": "before-hook-creation,hook-succeeded",
       "helm.sh/hook-weight": "1"
     },
-    "nodeSelector": {},
-    "podLabels": {},
     "rbac": {
       "annotations": {
         "helm.sh/hook": "post-install",
@@ -95,7 +71,6 @@
         "helm.sh/hook-weight": "-5"
       }
     },
-    "resources": {},
     "securityContext": {
       "runAsNonRoot": true
     },
@@ -108,17 +83,9 @@
       "automountServiceAccountToken": true,
       "create": true
     },
-    "timeout": "1m",
-    "tolerations": []
+    "timeout": "1m"
   },
-  "strategy": {},
-  "tolerations": [],
-  "volumeMounts": [],
-  "volumes": [],
   "webhook": {
-    "affinity": {},
-    "containerSecurityContext": {},
-    "extraArgs": [],
     "hostNetwork": false,
     "image": {
       "pullPolicy": "IfNotPresent",
@@ -131,8 +98,6 @@
       "successThreshold": 1,
       "timeoutSeconds": 1
     },
-    "nodeSelector": {},
-    "podLabels": {},
     "readinessProbe": {
       "failureThreshold": 3,
       "initialDelaySeconds": 5,
@@ -141,7 +106,6 @@
       "timeoutSeconds": 1
     },
     "replicaCount": 1,
-    "resources": {},
     "securePort": 10250,
     "securityContext": {
       "runAsNonRoot": true
@@ -150,11 +114,8 @@
       "automountServiceAccountToken": true,
       "create": true
     },
-    "serviceLabels": {},
     "serviceType": "ClusterIP",
-    "strategy": {},
     "timeoutSeconds": 10,
-    "tolerations": [],
     "url": {}
   }
 }

```

Reply:
@wallrj removing these empty attributes will cause a different render
example:
```diff
# Source: cert-manager/templates/webhook-deployment.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: cert-manager-webhook

...

                fieldPath: metadata.namespace
          resources:
-           {}
+           null
```

I think this is because `{{ toYaml .Values.resources | indent 12 }}` will print a different value for not-found values vs. empty values.